packages/node/test/manual/release-health/application-mode-session.js

```diff
@@ -0,0 +1,77 @@
+const Sentry = require('../../../dist');
+
+function assertSessions(actual, expected) {
+  if (JSON.stringify(actual) !== JSON.stringify(expected)) {
+    console.error('FAILED: Sessions do not match');
+    process.exit(1);
+  }
+}
+
+function constructStrippedSessionObject(actual) {
+  const { init, status, errors, release, did } = actual;
+  return { init, status, errors, release, did };
+}
+
+let remaining = 2;
+
+class DummyTransport {
+  sendEvent(event) {
+    return Promise.resolve({
+      status: 'success',
+    });
+  }
+  sendSession(session) {
+    if (remaining === 2) {
+      assertSessions(constructStrippedSessionObject(session),
+        {
+          init: true,
+          status: "ok",
+          errors: 1,
+          release: "1.1",
+          did: "ahmed",
+        }
+      )
+    }
+    else {
+      assertSessions(constructStrippedSessionObject(session),
+        {
+          init: false,
+          status: "exited",
+          errors: 1,
+          release: "1.1",
+          did: "ahmed",
+        }
+      )
+    }
+
+    --remaining;
+
+    if (!remaining) {
+      console.log('SUCCESS: All application mode sessions were sent to node transport as expected');
+    }
+    return Promise.resolve({
+      status: 'success',
+    });
+  }
+  close() {
+    return Promise.resolve({
+      status: 'success',
+    });
+  }
```

Comment:
Findings so far:
`process.on('exit')` will not function correctly because `endSession` is asynchronous and according to https://nodejs.org/api/process.html#process_event_exit

> Listener functions must only perform synchronous operations. The Node.js process will exit immediately after calling the 'exit' event listeners causing any additional work still queued in the event loop to be abandoned. In the following example, for instance, the timeout will never occur:

Possible Solutions
- https://stackoverflow.com/questions/40574218/how-to-perform-an-async-operation-on-exit
- Or using `beforeExit` event, however `beforeExit` has certain limitations
>The 'beforeExit' event is not emitted for conditions causing explicit termination, such as calling process.exit() or uncaught exceptions.
>
>The 'beforeExit' should not be used as an alternative to the 'exit' event unless the intention is to schedule additional work.

https://nodejs.org/api/process.html#process_event_beforeexit